include/aspect/initial_conditions/adiabatic_boundary.h

```diff
@@ -0,0 +1,107 @@
+   along with ASPECT; see the file doc/COPYING.  If not see
+   <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/initial_conditions/interface.h>
+#include <aspect/simulator.h>
+
+
+namespace aspect
+{
+  namespace InitialConditions
+  {
+    using namespace dealii;
+
+    /**
+     * A class that computes an initial temperature field based
+     * on a user-defined adiabatic boundary for the 3D ellipsoid
+     * chunk geometry model. It discretizes the model domain into
+     * two regions separated by an isotherm boundary below which
+     * the temperature increases adiabatically. Above the
+     * user-defined isotherm boundary the temperature linearly
+     * increases from a surface temperature (273.15 K or 0 degree C)
+     * to the isotherm (1673.15 K or 1600 degree C). The user
+     * defines the location of the isotherm boundary with an ascii
+     * data file with the format defined in the ASPECT manual. Note
+     * that the latitudinal and longitudinal bounds of the ascii input
+     * data file need to be at least 1 degree wider than the bounds
+     * you use to define the region of your ellipsoid chunk geometry.
+     *
+     * This plugin is developed by Tahiry Rajaonarison, D. Sarah Stamps,
+     * and Wolfgang Bangerth.
+     */
+    template <int dim>
+    class AdiabaticBoundary : public Interface<dim>
+    {
+      public:
+        /**
+         * Return the initial temperature as a function of position.
+         */
+
+        virtual
+        double initial_temperature (const Point<dim> &position) const;
+
+        /**
+         * Declare the parameters that this class needs.
+         */
+        static
+        void declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters above from the parameter file.
+         */
+        virtual
+        void parse_parameters (ParameterHandler &prm);
+
+      private:
+        std::vector<double>  latitudes_iso;
+        std::vector<double>  longitudes_iso;
+        std::vector<double>  depths_iso;
+        std::string isotherm_file_name;
+        double isotherm_temperature;
+        double surface_temperature;
+        double delta;
+
+        /**
+         * A function that reads the depth of the user-defined
+         * adiabatic boundary from an ascii data file and returns
+         * the value of the depth for each position.
+         */
+        double
+        get_isotherm_depth (const double latitude,
+                            const double longitude) const;
+
+        /**
+         * Return latitude and longitude from x,y and z in the WGS84
+         * Earth-Center Earth-Fixed (ECEF) reference frame.
+         */
+        std::pair<double, double>
+        lat_long_from_xyz_WGS84(const Point<3> &pos) const;
+
+        /**
+        * Return distance from the Earth's center to a point on the
+        * surface in the WGS84 ECEF reference frane.
+        */
+        double
+        radius_WGS84(const double &theta) const;
```

Comment:
For simple data types like double handing them over by reference does not gain an advantage and we usually do it by value. This line is not wrong, but removing the `&` will make it look more similar to the rest of the code.

Also could you specify in the description that this function will return the radius for *every* point of a given latitude? At first I was confused because you mention *a* point but only define a latitude.